APMrover2/APMrover2.cpp

```diff
@@ -245,7 +245,6 @@ void Rover::update_logging2(void)
 
     if (should_log(MASK_LOG_RC)) {
```

Comment:
This check has been lost.  I don't care.

Reply:
Peter, how do we handle logging once it's moved into a library.  For example I wonder if this will log while the vehicle is disarmed.

Reply:
fixed, makes sense to use the flag in the vehicle and it also logs at the same rate too. I've updated it so it's simplified